nakama/lib/src/models/session.dart

```diff
@@ -61,6 +61,31 @@ class Session with _$Session {
     );
   }
 
+  factory Session.restore({required String sessionToken,String? refreshToken}) {
+     final token = JwtDecoder.decode(sessionToken);
+     assert(token.containsKey('uid'));
+
+     int refreshExpireTime = 0;
+     if(refreshToken != null) {
+       final refreshTokenMap = JwtDecoder.decode(refreshToken!);
+       refreshExpireTime = refreshTokenMap['exp'] as int;
+     }
+
+    return Session(
+      token: sessionToken,
+      refreshToken: refreshToken,
+      created: false,
+      vars: token.containsKey('vars') ? token['vars'] as Map<String, String>? : {},
+      userId: token['uid'] as String,
+      expiresAt: DateTime.fromMillisecondsSinceEpoch(
+        (token['exp'] as int) * 1000,
```

Comment:
Why not just use the value directly instead of doing these conversions? (Same below.)

Reply:
Hi I'm not sure if I get what you mean but I did it just similar to the previous function of [session class](https://github.com/heroiclabs/nakama-dart/blob/main/nakama/lib/src/models/session.dart) which is `Session.fromApi` and `Session.fromDto` which the string tokens are converted into a map with dynamic values and cast according to their type.

Reply:
@garlen-javier The server returns the values as seconds rather than milliseconds -- could you change this code and the code you referenced to just use `token['exp'] as int` rather than doing the millisecond conversions? And same for the refresh token.

Reply:
Ok I get what you mean but in Dart you need those conversion to convert into DateTime and no directly way on doing it aside from using `DateTime.fromMicrosecondsSinceEpoch` or `DateTime.fromMillisecondsSinceEpoch` at which both you still need to multiply the `seconds` by 10000 or 1000.

Reply:
Ah I see, thanks.